src/v3/src/transformer/selectAuthenticator/utils.ts

```diff
@@ -167,57 +203,80 @@ const formatAuthenticatorOptions = (
   options: IdxOption[],
   step: string,
   isEnroll?: boolean,
-): AuthenticatorButtonElement[] => (
-  options.map((option: IdxOption, index: number) => {
-    const authenticatorKey = option.relatesTo?.key as string;
-    const id = getOptionValue(option.value as Input[], 'id')?.value;
-    const methodType = getOptionValue(option.value as Input[], 'methodType')?.value;
-    const enrollmentId = getOptionValue(option.value as Input[], 'enrollmentId')?.value;
-    const AUTHENTICATORS_WITH_METHOD_TYPE = [
-      AUTHENTICATOR_KEY.ON_PREM,
-      AUTHENTICATOR_KEY.RSA,
-    ];
-    const authenticator = option.relatesTo;
+): AuthenticatorButtonElement[] => {
+  const authenticatorOptionSet = new Set<string>();
+  return options
+    .filter((option: IdxOption) => {
+      if (isEnroll) {
+        return true;
+      }
+      // If webauthn enrollments > 1 just show one entry with a generic namne (first)
```

Comment:
Isn't this a problem during verification, though? If you only show one, it might not match the device the user wants to verify with.

Reply:
I wondered the same thing. 

[This](https://github.com/okta/okta-signin-widget/blob/f3f40ed2c66db24abafbbfcb0a5d712ad23d35e8/src/v2/view-builder/internals/FormInputFactory.ts#L48-L59) is the logic this mimics. 

Let me reach out to Aaron/SIW team to understand how that works more.

Reply:
Looks like it does work as we expect based on [this](https://okta.slack.com/archives/CAZH2MWEL/p1666210457353909) note from Vijet, but I also tested on v3 with this change and it also works. What I have noticed however, is that, it doesnt seem like this logic matters anymore (or maybe I haven't ran into the exact experience/scenario?). What I am seeing in both v2 and v3 is that when the authenticator options are passed to the Widgets from auth-js, even if we are enrolled in multiple webauthN factors it only ever gives one option back, and it is ALWAYS the correct one for the browser we are in. 

It seems like auth-js may be doing something on their end to determine the device we are on and surfaces the corresponding factor id to the widget. 
<img width="401" alt="Screen Shot 2022-10-20 at 5 03 19 PM" src="https://user-images.githubusercontent.com/97472729/197059934-7c326bcf-8da4-41bb-aff3-768c8916a006.png">
<img width="407" alt="Screen Shot 2022-10-20 at 5 14 22 PM" src="https://user-images.githubusercontent.com/97472729/197059938-1470c4ed-b838-41fb-8ba8-d5c4548681fd.png">

So now my question is, should I keep this logic, skip this test for v3 and remove this logic, or remove the logic in both v2 and v3 and remove the test?

Reply:
To unblock this PR, we can keep the logic. I think we can followup with a tech-debt JIRA to research further and remove the logic later in both widgets with SDK team's input.